packages/editor/src/components/inserter-sidebar/index.js

```diff
@@ -20,27 +20,48 @@ export default function InserterSidebar( {
 	closeGeneralSidebar,
 	isRightSidebarOpen,
 } ) {
-	const { insertionPoint, showMostUsedBlocks } = useSelect( ( select ) => {
-		const { getInsertionPoint } = unlock( select( editorStore ) );
-		const { get } = select( preferencesStore );
-		return {
-			insertionPoint: getInsertionPoint(),
-			showMostUsedBlocks: get( 'core', 'mostUsedBlocks' ),
-		};
-	}, [] );
+	const { inserterSidebarToggleRef, insertionPoint, showMostUsedBlocks } =
+		useSelect( ( select ) => {
+			const { getInserterSidebarToggleRef, getInsertionPoint } = unlock(
+				select( editorStore )
+			);
+			const { get } = select( preferencesStore );
+			return {
+				inserterSidebarToggleRef: getInserterSidebarToggleRef(),
+				insertionPoint: getInsertionPoint(),
+				showMostUsedBlocks: get( 'core', 'mostUsedBlocks' ),
+			};
+		}, [] );
 	const { setIsInserterOpened } = useDispatch( editorStore );
 
 	const isMobileViewport = useViewportMatch( 'medium', '<' );
 	const libraryRef = useRef();
 
+	// When closing the inserter, focus should return to the toggle button.
+	const closeInserterSidebar = useCallback( () => {
+		setIsInserterOpened( false );
+		inserterSidebarToggleRef.current?.focus();
+	}, [ inserterSidebarToggleRef, setIsInserterOpened ] );
+
+	const closeOnEscape = useCallback(
+		( event ) => {
+			if ( event.keyCode === ESCAPE && ! event.defaultPrevented ) {
+				event.preventDefault();
+				closeInserterSidebar();
+			}
+		},
+		[ closeInserterSidebar ]
+	);
+
 	return (
-		<div className="editor-inserter-sidebar">
+		// eslint-disable-next-line jsx-a11y/no-static-element-interactions
```

Comment:
We should address this in a future PR.

Reply:
The fix would be moving this interaction up to where the role for the region on the wrapper already is.

Reply:
> moving this interaction up

Why not do it here? Is it complex?

Reply:
I think it'll be better to do it when refactoring the list view and inserter to share the same sidebar.